### DISCUSSION TITLE
Release 8

### DISCUSSION THREAD

**Discussion by PeterH24x7** [PeterH24x7/-Solar-Analytics-integration-for-Home-Assistant-Energy-monitoring#25] [General]
Hi @everyone,

With help from @aaronpowell and @Roving-Ronin there are some minor fixes and tidy up to release 7, now available as a pre-release 8 version. Nothing is functionally new, but there are improvements regarding how the rest gets are triggered will make the integration a little more robust, timing optimal and consistent. See the inband comments of the code for more information.

The Solar Analytics standard, 3-phase and advance yaml files for release 8 are now in the main code tree. 

- [solar_analytics.yaml](https://github.com/PeterH24x7/-Solar-Analytics-integration-for-Home-Assistant-Energy-monitoring/blob/main/solar_analytics.yaml)
- [solar_analytics_3phase.yaml](https://github.com/PeterH24x7/-Solar-Analytics-integration-for-Home-Assistant-Energy-monitoring/blob/main/solar_analytics_3phase.yaml)
- [solar_analytics_advanced.yaml](https://github.com/PeterH24x7/-Solar-Analytics-integration-for-Home-Assistant-Energy-monitoring/blob/main/solar_analytics_advanced.yaml)

I'll let things run for another week or so, and if there are no complaints I'll publish the new release 8. Please test as you see fit and let me know if you have any problems. 

Chrs, Peter

**Comment by aaronpowell** [comment]
Just updated my instance to the new 3phase and it seems to be all good presently

**Comment by PeterH24x7** [comment]
Final miscellaneous change in main code tree includes trigger default value for non-numeric sa_site_id (in case such a thing exists). If you have already upgraded and it's working then there's no need to upgrade again. For details see [link](https://github.com/PeterH24x7/-Solar-Analytics-integration-for-Home-Assistant-Energy-monitoring/commit/2e7c8da58ddf7db34d4bbb6db05cdf445b2b55dc).

**Comment by saltydog256** [comment]
Thanks to Peter I now have two instances of Solar Analytics running on HA.  One 3-phase, the other single.  There are a few setup tricks in SA as there are specific default configurations to choose from, and there should be at least one consumption declared in each  SA monitor - not just things you want to monitor (my consumption was in the 3-phase SA).

**Comment by PeterH24x7** [comment]
Looks like there's nothing further to discuss on release 8. Release 9 currently in test. See other related discussions.
